src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -164,13 +164,16 @@ protected override void ExecuteCore()
 
                 if (EmitSymbols)
                 {
-                    if (IsTargetWindows && hasValidDiaSymReaderLib)
+                    if (IsTargetWindows)
                     {
-                        outputPDBImage = Path.ChangeExtension(outputR2RImage, "ni.pdb");
-                        outputPDBImageRelativePath = Path.ChangeExtension(outputR2RImageRelativePath, "ni.pdb");
-                        crossgen1CreatePDBCommand = $"/CreatePDB \"{Path.GetDirectoryName(outputPDBImage)}\"";
+                        if (hasValidDiaSymReaderLib)
+                        {
+                            outputPDBImage = Path.ChangeExtension(outputR2RImage, "ni.pdb");
+                            outputPDBImageRelativePath = Path.ChangeExtension(outputR2RImageRelativePath, "ni.pdb");
+                            crossgen1CreatePDBCommand = $"/CreatePDB \"{Path.GetDirectoryName(outputPDBImage)}\"";
+                        }
                     }
-                    else if (IsTargetLinux)
+                    else if (!_crossgen2IsVersion5 || IsTargetLinux)
```

Comment:
Should this be
```suggestion
                    else if (ReadyToRunUseCrossgen2 && !_crossgen2IsVersion5 || IsTargetLinux)
```
as we cannot generate symbols on macOS using the old cressgen?

Reply:
Hmm, how come it doesn't hit the condition <code>!_crossgen2IsVersion5</code>?

Reply:
I think I'm starting to get it, let me try it out...

Reply:
If we are using the old crossgen on macOS, `_crossgen2IsVersion5` is `false`, we execute this branch and use the unsupported `/CreatePerfMap` switch.

Reply:
Thanks Anton for explaining, I have modified the source file based on your suggestion in the 9th commit.